diff --git a/cmake/FindGASNet.cmake b/cmake/FindGASNet.cmake
index 9ac197ea0..ffc0f29da 100644
--- a/cmake/FindGASNet.cmake
+++ b/cmake/FindGASNet.cmake
@@ -277,7 +277,7 @@ if(NOT GASNet_FOUND AND NOT TARGET GASNet::GASNet)
   if(GASNet_FOUND)
     # Reorder preferred conduits to place local preferences first
     set(GASNet_ALL_PREFERRED_CONDUITS
-      "mxm;psm;gemini;aries;pami;ibv;shmem;portals4;ofi;mpi;udp;smp;ucx")
+      "mxm;psm;gemini;aries;pami;ibv;shmem;portals4;ofi;ofi-slingshot11;mpi;udp;smp;ucx")
     set(GASNet_PREFERRED_CONDUITS "${GASNet_ALL_PREFERRED_CONDUITS}"
       CACHE STRING "")
     mark_as_advanced(GASNet_PREFERRED_CONDUITS)
@@ -325,13 +325,15 @@ if(GASNet_FOUND AND NOT TARGET GASNet::GASNet)
     message(FATAL_ERROR "Invalid GASNet_THREADING setting.  Valid options are: ${GASNet_THREADINGS}")
   endif()
 
-  if(NOT TARGET GASNet::${GASNet_CONDUIT}-${GASNet_THREADING})
-    message(FATAL_ERROR "Unable to use selected CONDUIT-THREADING combination: ${GASNet_CONDUIT}-${GASNet_THREADING}")
+  string(FIND "${GASNet_CONDUIT}" "-" first_dash_idx)
+  string(SUBSTRING "${GASNet_CONDUIT}" 0 ${first_dash_idx} GASNet_CONDUIT_SHORT)
+  if(NOT TARGET GASNet::${GASNet_CONDUIT_SHORT}-${GASNet_THREADING})
+	  message(FATAL_ERROR "Unable to use selected CONDUIT-THREADING combination: ${GASNet_CONDUIT_SHORT}-${GASNet_THREADING}")
   endif()
-  message(STATUS "GASNet: Using ${GASNet_CONDUIT}-${GASNet_THREADING}")
+  message(STATUS "GASNet: Using ${GASNet_CONDUIT_SHORT}-${GASNet_THREADING}")
   add_library(GASNet::GASNet INTERFACE IMPORTED)
   set_target_properties(GASNet::GASNet PROPERTIES
     INTERFACE_COMPILE_DEFINITIONS GASNETI_BUG1389_WORKAROUND=1
-    INTERFACE_LINK_LIBRARIES GASNet::${GASNet_CONDUIT}-${GASNet_THREADING}
+    INTERFACE_LINK_LIBRARIES GASNet::${GASNet_CONDUIT_SHORT}-${GASNet_THREADING}
   )
 endif()
diff --git a/runtime/CMakeLists.txt b/runtime/CMakeLists.txt
index 634504887..2e16cde3a 100644
--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -395,7 +395,7 @@ if(REALM_USE_GASNET1 OR REALM_USE_GASNETEX)
     if(BUILD_SHARED_LIBS)
       # get the list of libraries and filter only those that are not in the
       #  (gasnet) install dir - the gasnet ones are sucked into librealm.so
-      get_target_property(gasnet_ext_libs GASNet::${GASNet_CONDUIT}-par INTERFACE_LINK_LIBRARIES)
+      get_target_property(gasnet_ext_libs GASNet::${GASNet_CONDUIT_SHORT}-par INTERFACE_LINK_LIBRARIES)
       list(FILTER gasnet_ext_libs EXCLUDE REGEX "^${GASNet_INSTALL_DIR}")
       set(GASNET_LINK_LIBRARIES_FOR_INSTALL ${gasnet_ext_libs} PARENT_SCOPE)
       # cmake won't include GASNet::GASNet in the exported link libraries
