diff --git a/install.py b/install.py
index 25bf4e7..38f9cec 100755
--- a/install.py
+++ b/install.py
@@ -452,7 +452,7 @@ def install(
     if ucx_dir:
         cmake_flags += [f"-DUCX_ROOT={ucx_dir}"]
     if conduit:
-        cmake_flags += [f"-DGASNet_CONDUIT={conduit}"]
+        cmake_flags += [f"-DGASNet_CONDUIT={conduit}"] #.split('-')[0]}"]
     if cuda_dir:
         cmake_flags += [f"-DCUDAToolkit_ROOT={cuda_dir}"]
     if thrust_dir:
@@ -643,7 +643,7 @@ def driver():
         # TODO: To support UDP conduit, we would need to add a special case on
         # the legate launcher.
         # See https://github.com/nv-legate/legate.core/issues/294.
-        choices=["ibv", "ucx", "aries", "mpi"],
+        choices=["ibv", "ucx", "aries", "mpi", "udp", "ofi-slingshot11"],
         default=os.environ.get("CONDUIT"),
         help="Build Legate with specified GASNet conduit.",
     )
diff --git a/scripts/generate-conda-envs.py b/scripts/generate-conda-envs.py
index 9222e2b..869cc33 100755
--- a/scripts/generate-conda-envs.py
+++ b/scripts/generate-conda-envs.py
@@ -29,7 +29,7 @@ def V(version: str) -> tuple[int, ...]:
     return tuple(int(x) for x in padded_version)
 
 
-class SectionConfig(Protocol):
+class SectionConfig:
     header: str
 
     @property
